### PR TITLE
Update `uefi` dependency to `v0.20`

### PR DESCRIPTION
**Pull request by kennystrawnmusic**
The old version (0.18.0) of the UEFI crate that bootloader 0.11.3 depends on has code that throws a myriad of future incompatibility/deprecation warnings when compilation is attempted ― deprecation warnings that are highlighted in the rust-lang/rust#107457 tracking issue. I therefore created this pull request that both fixes the version of UEFI depended on (0.20 doesn't have the same problems) and bumps the version of bootloader to 0.11.4 to ensure that this fix gets published to crates.io immediately after it's implemented.

### CODE REVIEW

File: uefi/src/main.rs
Comment (phil-opp) [inline uefi/src/main.rs:139]: This is no longer needed because of https://github.com/rust-osdev/uefi-rs/pull/653. We should remove this mapping, not just silence the dead code warning.

File: Cargo.toml
Comment (phil-opp) [inline Cargo.toml:34]: Please don't bump the version here. You can request a new release in the comments, but the decision when to releases is up to the maintainers. Also, we doing the release should be done in a separate PR anyway because it requires additional steps (e.g. changelog creation).

File: tests/runner/src/lib.rs
Comment (phil-opp) [inline tests/runner/src/lib.rs:5]: Instead of silencing the warning, we should move this constant into `run_qemu`.

File: tests/runner/src/lib.rs
Comment (phil-opp) [inline tests/runner/src/lib.rs:16]: Instead of silencing the warning, we should move this constant into `run_qemu`.

File: tests/runner/src/lib.rs
Comment (phil-opp) [inline tests/runner/src/lib.rs:104]: We should use the proper `cfg` flags here instead of just silencing the dead code warning:

```suggestion
#[cfg(any(feature = "uefi", feature = "bios"))]
```

**Review by phil-opp — COMMENTED** [review]
Thanks for the PR! The update `uefi v0.20` looks good. I left some comments inline. Once they're resolved, we can merge this.

**Comment by kennystrawnmusic** [comment]
> Thanks for the PR! The update `uefi v0.20` looks good. I left some comments inline. Once they're resolved, we can merge this.

Going to do this tonight (Pacific time) — with family at the moment.

**Comment by kennystrawnmusic** [comment]
> Thanks for the PR! The update `uefi v0.20` looks good. I left some comments inline. Once they're resolved, we can merge this.

Alright, just reviewed them. Sorry it took longer than expected; been very busy lately.

**Review by phil-opp — APPROVED** [review]
Thanks!

**Pull request merged**
